sql/plan/analyze.go

```diff
@@ -0,0 +1,205 @@
+package plan
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/mysql_db"
+)
+
+// TODO: do i need additional database argument?
+type Analyze struct {
+	db   sql.Database
+	tbls []sql.Node
+}
+
+func NewAnalyze(db sql.Database, tbls []sql.Node) *Analyze {
+	return &Analyze{
+		db:   db,
+		tbls: tbls,
+	}
+}
+
+// Schema implements the interface sql.Node.
+// TODO: should be |Tables|Op|Msg_type|Msg_text|
+func (n *Analyze) Schema() sql.Schema {
+	return sql.OkResultSchema
+}
+
+// String implements the interface sql.Node.
+func (n *Analyze) String() string {
+	tblNames := make([]string, len(n.tbls))
+	for i, tbl := range n.tbls {
+		switch resTbl := tbl.(type) {
+		case *ResolvedTable:
+			tblNames[i] = resTbl.Name()
+		case *UnresolvedTable:
+			tblNames[i] = resTbl.Name()
+		case *Exchange:
+			tblNames[i] = resTbl.Child.String()
+		}
+	}
+	return fmt.Sprintf("Analyze table %s", strings.Join(tblNames, ", "))
+}
+
+// Database implements the interface sql.Databaser.
+func (n *Analyze) Database() sql.Database {
+	return n.db
+}
+
+// WithDatabase implements the interface sql.Databaser.
+func (n *Analyze) WithDatabase(db sql.Database) (sql.Node, error) {
+	nn := *n
+	nn.db = db
+	return &nn, nil
+}
+
+// Resolved implements the Resolvable interface.
+func (n *Analyze) Resolved() bool {
+	_, ok := n.db.(sql.UnresolvedDatabase)
+	for _, tbl := range n.tbls {
+		if !tbl.Resolved() {
+			return false
+		}
+	}
+	return !ok
+}
+
+// Children implements the interface sql.Node.
+func (n *Analyze) Children() []sql.Node {
+	return n.tbls
+}
+
+// WithChildren implements the interface sql.Node.
+func (n *Analyze) WithChildren(children ...sql.Node) (sql.Node, error) {
+	// Deep copy kids
+	newChildren := make([]sql.Node, len(children))
+	for i, child := range children {
+		newChildren[i] = child
+	}
+
+	nn := *n
+	nn.tbls = newChildren
+	return &nn, nil
+}
+
+// CheckPrivileges implements the interface sql.Node.
+func (n *Analyze) CheckPrivileges(ctx *sql.Context, opChecker sql.PrivilegedOperationChecker) bool {
+	return true
+}
+
+// RowIter implements the interface sql.Node.
+func (n *Analyze) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// Assume table is in current database
+	database := ctx.GetCurrentDatabase()
+	if database == "" {
+		return nil, sql.ErrNoDatabaseSelected.New()
+	}
+
+	// Access mysql db, which is called GrantTables for now
+	mysql, ok := n.db.(*mysql_db.MySQLDb)
+	if !ok {
+		return nil, sql.ErrDatabaseNotFound.New("mysql")
+	}
+
+	// Get column statistics table
+	colStatsTableData := mysql.ColumnStatisticsTable().Data()
+
+	// Go through each table in node
+	for _, tbl := range n.tbls {
+		// Check if table was resolved
+		var resTbl *ResolvedTable
+		switch v := tbl.(type) {
+		case *ResolvedTable:
```

Comment:
i'm not sure if they can reach here, but `IndexedTableAccess`, `DeferredAsofTable`, and maybe a couple others are valid table types. A prepared enginetest would probably break, for example.